apis/v1alpha2/targetallocator_types.go

```diff
@@ -35,8 +39,12 @@ type TargetAllocatorSpec struct {
 	// +optional
 	// +kubebuilder:default:=relabel-config
 	FilterStrategy TargetAllocatorFilterStrategy `json:"filterStrategy,omitempty"`
-	// ServiceAccount indicates the name of an existing service account to use with this instance. When set,
-	// the operator will not automatically create a ServiceAccount for the TargetAllocator.
+	// ScrapeConfigs define static Prometheus scrape configurations for the target allocator.
+	// To use dynamic configurations from ServiceMonitors and PodMonitors, see the PrometheusCR section.
+	// +optional
+	// +listType=atomic
+	ScrapeConfigs []ScrapeConfig `json:"scrapeConfigs,omitempty"`
```

Comment:
this makes sense to add, but I also wonder if having the collector selector above should hint the operator to pull the configmap for that collector? Is that too difficult / cumbersome / confusing?

Reply:
That should also work, the way I imagined it is that the operator would set this field based on the annotation. I'm not actually sure what the idiomatic solution to this kind of problem is, but I'm defaulting to being more explicit.

Reply:
Could we at least put a link to p8s docs where people can find how to define scrapeconfigs?